app/models/compute_attribute.rb

```diff
@@ -18,6 +18,10 @@ def method_missing(method, *args, &block)
     raise Foreman::Exception.new(N_('%s is an unknown attribute'), method)
   end
 
+  def respond_to?(method)
```

Comment:
Ah good catch, I didn't realize method_missing implementation deserves some refactoring as well. The whole idea is that we pretend that ComputeResource instance has methods based on vm_attrs keys. Therefore all other introspection helpers should work, someone might want to know whether this duck quacks. Without the patch, 

```ruby
ComputeResource.new.respond_to?(:cluster) # => false
ComputeResource.new.cluster # => nil
```

while with patch

```ruby
ComputeResource.new.respond_to?(:cluster) # => true
ComputeResource.new.respond_to?(:clusterx) # => false
```
I've changed it to `respond_to_missing?` so [other calls](https://robots.thoughtbot.com/always-define-respond-to-missing-when-overriding) works as well

> Notice vm_attrs might not always be defined

I think it is always defined, since it's DB column. I guess the hack is needed because active record also uses `method_missing` for these accessors.